mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksStateFactory.kt

```diff
@@ -77,21 +78,29 @@ internal fun <VM : MavericksViewModel<S>, S : MavericksState> createStateFromCon
             if (Modifier.isPublic(stateClass.modifiers)) {
                 stateClass.newInstance()
             } else {
-                stateClass.constructors.firstOrNull { it.parameterCount == 0 }?.let { c ->
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    stateClass.constructors.firstOrNull { it.parameterCount == 0 }
+                } else {
+                    // In older API's without access to parameter count, assume the last constructor
+                    // is the empty one, as generated kotlin byte code shows the last constructor to be the one using all default values.
+                    stateClass.constructors.lastOrNull()
+                }?.let { c ->
                     c.isAccessible = true
                     c.newInstance() as? S
                 }
             }
         } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
             // Throw this exception if something we explicitly tried failed.
-            throw java.lang.IllegalStateException("Failed to create initial state!", e)
+            throw IllegalStateException("Failed to create initial state!", e)
```

Comment:
`error(...)`?

Reply:
ah, can't use that with a nested exception actually